R/type-data-frame.R

```diff
@@ -27,12 +33,15 @@ new_data_frame <- function(x = list(), n = NULL, ..., class = character()) {
     names(x) <- character()
   }
 
-  structure(
-    x,
+  new_attributes <- list(
     ...,
     class = c(class, "data.frame"),
     row.names = .set_row_names(n)
   )
+
+  attributes(x) <- c(attributes(x), new_attributes)
```

Comment:
I think a constructor should overwrite existing attributes?

Reply:
`structure()` doesn't strip existing attributes, and that's how we get the `names()` off of `x` for the new data frame, but I can alter it to just be like `c(list(names = names(x), new_attributes)` (which feels appropriate because right now I think the `tzone` attribute is coming along too, and it shouldn't)